src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -385,4 +386,68 @@ List<ExternalObjectDirectoryEntity> findByExternalLocationTypeAndUpdateRetention
 
 
     List<ExternalObjectDirectoryEntity> findByManifestFile(String manifestName);
+
+    @Query("""
+               SELECT eod
+               FROM ExternalObjectDirectoryEntity eod
+               JOIN TranscriptionDocumentEntity t ON eod.transcriptionDocumentEntity = t
+               WHERE t.retainUntilTs < :maxRetentionDate
+               AND t.isDeleted = false
+               AND eod.externalLocationType.id in (1,2,3)
+               AND EXISTS (
+                SELECT 1 FROM ExternalObjectDirectoryEntity eod2
+                WHERE eod2.externalLocationType.id = 3
+                AND eod2.status.id = 2
+                AND eod2.transcriptionDocumentEntity = eod.transcriptionDocumentEntity
+               )
+        """)
+    List<ExternalObjectDirectoryEntity> findExpiredTranscriptionDocuments(OffsetDateTime maxRetentionDate, Limit batchSize);
+
+    @Query("""
+               SELECT eod
+               FROM ExternalObjectDirectoryEntity eod
+               JOIN AnnotationDocumentEntity a ON eod.annotationDocumentEntity = a
+               WHERE a.retainUntilTs < :maxRetentionDate
+               AND a.isDeleted = false
+               AND eod.externalLocationType.id in (1,2,3)
```

Comment:
We could accept the entity types see findMatchingExternalObjectDirectoryEntityByLocation

Reply:
This is a performance thing, in this instance we don't care about the all the data within the LocationTypeEnity so we don't need to fetch the object

Reply:
ok fine with that maybe in future we could pass though ExternalLocationTypeEnum